frontend/cypress/integration/common/service_details.ts

```diff
@@ -0,0 +1,103 @@
+import { And, Given, Then, When } from 'cypress-cucumber-preprocessor/steps';
+import {checkHealthIndicatorInTable, checkHealthStatusInTable, getCellsForCol, getColWithRowText} from './table';
+import { ensureKialiFinishedLoading } from './transition';
+
+const NAMESPACE = 'bookinfo';
+const SERVICE = 'productpage';
+const URL = '/console/namespaces/'+NAMESPACE+'/services/'+SERVICE+'?refresh=0';
+const tracingDotQuery = '[style*="fill: var(--pf-global--palette--blue-200)"][style*="stroke: transparent"]';
+
+function openTab(tab: string) {
+    cy.get('.pf-c-tabs__list').should('be.visible').contains(tab).click();
+}
+
+Given('a service in the cluster with a healthy amount of traffic', function () {
+    this.targetNamespace = NAMESPACE;
+    this.targetService = SERVICE;
+});
+
+Then('user sees a list with content {string}', (tab: string) => {
+    cy.get('.pf-c-tabs__list').contains(tab);
+});
+
+Then('user sees the actions button', () => {
+    cy.getBySel('wizard-actions').should('be.visible').click();
+    cy.getBySel('wizard-actions').siblings().contains("Request Routing");
+});
+
+Then('user sees {string} details information for service', (name: string) => {
+    cy.get('#ServiceDescriptionCard').within(() => {
+        cy.get('#pfbadge-S').parent().parent().contains(name); // Service
+        cy.get('#pfbadge-A').parent().parent().contains(name); // App
+        cy.get('#pfbadge-W').parent().parent().contains(name + '-v1'); // Workload
```

Comment:
one minor thing as well please, add version in params, so this method could be used for all services